zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterServer.java

```diff
@@ -649,13 +650,21 @@ public Object jobRun() throws Throwable {
         }
         // put result into resource pool
         if (resultMessages.size() > 0) {
-          int lastMessageIndex = resultMessages.size() - 1;
-          if (resultMessages.get(lastMessageIndex).getType() == InterpreterResult.Type.TABLE) {
+          final int lastMessageIndex = resultMessages.size() - 1;
+          final InterpreterResultMessage lastMessage = resultMessages.get(lastMessageIndex);
+          if (lastMessage.getType() == InterpreterResult.Type.TABLE) {
             context.getResourcePool().put(
                 context.getNoteId(),
                 context.getParagraphId(),
                 WellKnownResourceName.ZeppelinTableResult.toString(),
-                resultMessages.get(lastMessageIndex));
+                lastMessage
+            );
+            context.getResourcePool().put(
+                context.getNoteId(),
+                context.getParagraphId(),
+                WellKnownResourceName.ZeppelinTableType.toString(),
+                new MessageColumnTypes(lastMessage.getColumnTypes())
```

Comment:
Shouldn't the lastMessage already include the columns types ? It is not necessary to put the columns types into ResourcePool again.

Reply:
In that case we can't see type information in appropriate way:
![screenshot from 2018-07-12 12-11-27](https://user-images.githubusercontent.com/16215034/42624024-0dbb192c-85cd-11e8-95fc-37a31f64b508.png)

Reply:
I am not sure what do you mean `nice type info`. It is issue of py4j. That we can only get JavaObject via py4j.